### PR TITLE
LEARNER-4703: Adds in skeleton view and tests for Program Record page

### PR DESCRIPTION
**Pull request by jlajoie**
screen shot 2018-05-23 at 11 42 19 am
screen shot 2018-05-23 at 12 02 33 pm

### CODE REVIEW

File: credentials/static/components/ProgramRecord.jsx [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        {this.renderBackLink()}
+        {this.renderShareButton()}
+      </section>
+    );
+  }
+
+  static renderProgramName(data) {
+    return (
+      <div name="program-name" className={['hd-3', 'flex-1'].join(' ')}>
+        {data.name} {gettext('Transcript')}
```
[the remark this answers is not in the document]
Comment (jlajoie) [reply credentials/static/components/ProgramRecord.jsx]: Got a bit too literal there with the mockups sorry!

I'll investigate the interpolate work, there should be an easy way to do this and we'll definitely need it elsewhere.

File: credentials/static/components/ProgramRecord.jsx [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        <div className={['learner-name']}>
+          {gettext('Name')} {data.full_name}
+        </div>
+        <div className={['learner-username']}>
+          {gettext('edX User ID')} {data.username}
+        </div>
+        <div className={['learner-email']}>
+          {gettext('Email')} {data.email}
+        </div>
+      </section>
```
[the remark this answers is not in the document]
Comment (jlajoie) [reply credentials/static/components/ProgramRecord.jsx]: Sure absolutely, thought it was weird to do all of our styling via tables like it's 1999, but I can see why this works for that

[the remark this answers is not in the document]
Comment (mikix) [reply credentials/static/components/ProgramRecord.jsx]: Oh well a non-table way would work too?  I dunno, I was just suggesting we get it closer to the mockup a bit, but this is more in the visual realm, which we can leave to epic 3.

[the remark this answers is not in the document]
Comment (jlajoie) [reply credentials/static/components/ProgramRecord.jsx]: With the expected mobile view I'll just re-use the FoldingTable

File: credentials/static/components/ProgramRecord.jsx [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+      <div name="program-name" className={['hd-3', 'flex-1'].join(' ')}>
+        {data.name} {gettext('Transcript')}
+      </div>
+    );
+  }
+
+  static renderSchoolName(data) {
+    return (
+      <div name="school-name" className={['hd-3']}>
+        {gettext('edX |')} {data.school}
```
[the remark this answers is not in the document]
Comment (mikix) [reply credentials/static/components/ProgramRecord.jsx]: site_configuration.platform_name

File: credentials/static/components/ProgramRecord.jsx [hunk 4]
```diff
@@ -0,0 +1,157 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import { Button } from '@edx/paragon';
+
+import FoldingTable from './FoldingTable';
+
```
Comment (bderusha) [inline credentials/static/components/ProgramRecord.jsx]: naive style question, are the lines in between each import purposeful?

File: credentials/templates/programs.html [hunk 5]
```diff
@@ -0,0 +1,28 @@
+{% extends 'base.html' %}
+{% load html %}
+{% load i18n %}
+{% load render_bundle from webpack_loader %}
+
+{% block title %}
+  {% trans "{program} Record" as tmsg %}
+  {% interpolate_html tmsg program=program_name as tmsg %}
```
Comment (bderusha) [inline credentials/templates/programs.html]: NIT: I think for translators, calling this `program_name` would be better than just `program`

File: credentials/static/components/ProgramRecord.jsx (hunk 4 above)
Comment (jlajoie) [reply credentials/static/components/ProgramRecord.jsx]: Not positive on what the expected style is... but was stolen shamelessly from elsewhere

File: credentials/apps/records/views.py [hunk 6]
```diff
@@ -44,3 +44,88 @@ def dispatch(self, request, *args, **kwargs):
...
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        record = self._get_record()
+        context.update({
+            'child_templates': {
+                'footer': self.select_theme_template(['_footer.html']),
+                'header': self.select_theme_template(['_header.html']),
+            },
+            'record': json.dumps(self._get_record(), sort_keys=True),
```
[the remark this answers is not in the document]
Comment (jlajoie) [reply credentials/apps/records/views.py]: None, updated

**Pull request merged**